api/test/e2e/oidc/oidc_test.go

```diff
@@ -0,0 +1,227 @@
+		ginkgo.It("should return status-code 302", func() {
+			gomega.Expect(accessOidcLogin()).To(gomega.Equal(http.StatusFound))
+		})
+	})
+	ginkgo.Context("test apisix/admin/oidc/callback", func() {
+		ginkgo.It("should return status-code 200", func() {
+			gomega.Expect(accessOidcCallback()).To(gomega.Equal(http.StatusOK))
+		})
+	})
+
+	ginkgo.Context("access apisix/admin/routes with cookie", func() {
+		ginkgo.It("should return status-code 200", func() {
+			gomega.Expect(accessRoutesWithCookie(true)).To(gomega.Equal(http.StatusOK))
+		})
+	})
+
+	ginkgo.Context("access apisix/admin/oidc/logout with cookie", func() {
+		ginkgo.It("should return status-code 200", func() {
+			gomega.Expect(accessOidcLogoutWithCookie(true)).To(gomega.Equal(http.StatusOK))
+		})
+	})
+
+	ginkgo.Context("access apisix/admin/routes with invalid cookie", func() {
+		ginkgo.It("should return status-code 401", func() {
+			gomega.Expect(accessRoutesWithCookie(false)).To(gomega.Equal(http.StatusUnauthorized))
+		})
+	})
+
+	ginkgo.Context("access apisix/admin/oidc/logout with invalid cookie", func() {
+		ginkgo.It("should return status-code 403", func() {
+			gomega.Expect(accessOidcLogoutWithCookie(false)).To(gomega.Equal(http.StatusForbidden))
+		})
+	})
+})
+
+func AccessOidcLogin() http.Response {
+	var req *http.Request
+	var resp *http.Response
+	var Client = &http.Client{
+		CheckRedirect: func(req *http.Request, via []*http.Request) error {
+			return http.ErrUseLastResponse
+		},
+	}
+
+	req, _ = http.NewRequest("GET", "http://127.0.0.1:9000/apisix/admin/oidc/login", nil)
+	resp, _ = Client.Do(req)
+
+	// return status-code
+	return *resp
+}
+
+func accessOidcLogin() int {
+	// access apisix/admin/oidc/login
+	resp := AccessOidcLogin()
+	return resp.StatusCode
+}
+
+func createUser(authenticationUrl string) string {
+	u, _ := url.Parse(authenticationUrl)
+	client := gocloak.NewClient("http://" + u.Host)
+	ctx := context.Background()
+	token, _ := client.LoginAdmin(ctx, "admin", "admin", "master")
+
+	username := GetRandomString(3)
+	user := gocloak.User{
+		FirstName: gocloak.StringP(GetRandomString(3)),
+		LastName:  gocloak.StringP(GetRandomString(3)),
+		Email:     gocloak.StringP(GetRandomString(3)),
+		Enabled:   gocloak.BoolP(true),
+		Username:  gocloak.StringP(username),
+	}
+
+	id, _ := client.CreateUser(ctx, token.AccessToken, "master", user)
+
+	_ = client.SetPassword(ctx, token.AccessToken, id, "master", "123456", false)
+	return username
+}
+
+func accessOidcCallback() int {
+	var authenticationUrl string
+	var loginUrl string
+	var req *http.Request
+	var resp *http.Response
+	var Client http.Client
+
+	// access apisix/admin/oidc/login to get the authentication-url
+	*resp = AccessOidcLogin()
+	authenticationUrl = resp.Header.Get("Location")
+
+	// create a user
+	username := createUser(authenticationUrl)
+
+	// access the authentication-url
+	req, _ = http.NewRequest("GET", authenticationUrl, nil)
+	resp, _ = Client.Do(req)
```

Comment:
```suggestion
	resp, err = Client.Do(req)
```

gomega.Expect(err).ShouldNot(gomega.HaveOccurred(), "do request")